dsa/writer.go

```diff
@@ -0,0 +1,35 @@
+package dsa
+
+import (
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+// DSAWriter is used to write the default DSA to the request (req.regs.ext.dsa)
+type Writer struct {
+	Config      *config.AccountDSA
+	GDPRInScope bool
+}
+
+// Write sets the default DSA object on the request at regs.ext.dsa if it is
+// defined in the account config and it is not already present on the request
+func (dw Writer) Write(req *openrtb_ext.RequestWrapper) error {
```

Comment:
I don't know if this is a stronger version of the function, but here's a version where we combine all of the conditions that result in us returning nil, rather than breaking them all up:

```
func (dw Writer) Write(req *openrtb_ext.RequestWrapper) error {
	if req == nil || getReqDSA(req) != nil || dw.Config == nil || dw.Config.DefaultUnpacked == nil ||
		(dw.Config.GDPROnly && !dw.GDPRInScope) {
		return nil
	}

	regExt, err := req.GetRegExt()
	if err != nil {
		return err
	}

	regExt.SetDSA(dw.Config.DefaultUnpacked)
	return nil
}
```
What do you think?

Reply:
I think this is really a matter of preference. I see your perspective that there are multiple conditionals that result in the same return value, though my tendency is to break complex conditionals into logical groupings because I think it is easier to digest and instills confidence in the intended behavior. I also think it is easier to scan code vertically. In that case, at a minimum the first two conditionals could be grouped as such: `if req == nil || getReqDSA(req) != nil {`.
@SyntaxNode do you feel strongly either way here?

Reply:
Yeah I'm thinking your latest update looks great